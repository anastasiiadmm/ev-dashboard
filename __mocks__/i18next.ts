import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';

import translationEN from '~/shared/locales/translationEN.json';
import translationRU from '~/shared/locales/translationRU.json';
import translationKY from '~/shared/locales/translationKY.json';

i18n
  .use(initReactI18next)
  .init({
    resources: {
      en: { translation: translationEN },
      ru: { translation: translationRU },
      ky: { translation: translationKY },
    },
    lng: 'ru',
    fallbackLng: 'ru',
    interpolation: {
      escapeValue: false,
    },
  });

export default i18n;
