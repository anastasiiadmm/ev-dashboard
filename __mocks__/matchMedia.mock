window.matchMedia = window.matchMedia || function(query) {
  return {
    matches: false,
    addListener: function() {},
    removeListener: function() {},
    media: query,
    onchange: null,
    dispatchEvent: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
  };
};
