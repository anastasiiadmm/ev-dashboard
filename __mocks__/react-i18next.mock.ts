jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: (str) => str,
    i18n: {
      changeLanguage: () => new Promise(() => {}),
    },
  }),
  initReactI18next: {
    type: '3rdParty',
    init: () => {},
  },
  Trans: ({ children }) => children,
}));
