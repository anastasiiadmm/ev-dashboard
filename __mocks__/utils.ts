jest.mock('mobx', () => ({
  ...jest.requireActual('mobx'),
  observable: jest.fn(() => ({})),
}));
