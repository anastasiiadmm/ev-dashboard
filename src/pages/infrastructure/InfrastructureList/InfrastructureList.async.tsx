import { lazy } from 'react';

export const InfrastructureListAsync = lazy(() => import('./InfrastructureList'));
