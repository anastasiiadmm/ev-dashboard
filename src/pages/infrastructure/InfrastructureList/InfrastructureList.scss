@use '../../../app/styles/variables' as variables;

.InfrastructureList {
  &__search-pagination-block {
    box-sizing: border-box !important;
    z-index: 999;
    width: calc(100% + 40px);
    margin: 0;
    background: #ffffff;
    padding: 16px 16px 0 16px;
    border-top: 1px solid variables.$gray-bg-color;
    border-left: 1px solid variables.$gray-bg-color;
  }

  &__table-block {
    width: 100%;
    position: relative;
    margin: 20px;
    display: flex;
    flex-direction: column;
  }

  &__add-block {
    display: flex;
    align-self: self-end;
  }

  &__button-style {
    display: flex;
    align-self: flex-end;
    align-items: center;
    width: 237px;
    margin-bottom: 18px;
  }

  &__pagination-block {
    margin-top: 16px;
    position: absolute;
    bottom: 0;
  }

  &__tooltip-style {
    margin: 0;
    display: flex;
    align-items: center;
    gap: 6px;
  }

  &__info {
    display: flex;
    align-items: center;
    gap: 8px;
  }

  &__tags-block {
    display: flex;
    align-items: center;
    gap: 12px;
  }

  &__infrastructure {
    background-color: variables.$light-blue;
    border-radius: 8px;
    display: inline-block;
    padding: 5px 7px;
  }

  &__icon-infrastructure {
    width: 13px;
    vertical-align: text-bottom;
    margin-right: 4px;
  }

  &__center-block {
    display: block;
    text-align: center;
    margin: 0 auto;
  }

  &__add-button {
    display: block;
    text-align: center;
    background: variables.$active-link-menu-bg !important;
    border-color: variables.$active-link-menu-bg !important;
  }

  &__delete-button {
    display: block;
    text-align: center;
    background: variables.$light-red-bg !important;
    border-color: variables.$light-red-bg !important;
  }

  &__deactivate-button {
    background-color: variables.$white-color !important;
    border-color: variables.$white-color !important;
    color: variables.$red-bg;
  }

  &__activate-button {
    background-color: variables.$white-color !important;
    border-color: variables.$white-color !important;
    color: variables.$link-turquoise-color;
  }

  .ant-input-search > .ant-input-group > .ant-input-group-addon:last-child {
    inset-inline-start: 1px;
  }

  .ant-input-affix-wrapper {
    background-color: variables.$gray-bg-color;
    border-color: variables.$gray-bg-color;
  }

  .ant-input {
    background-color: variables.$gray-bg-color;
    border-color: variables.$gray-bg-color;
  }
}
