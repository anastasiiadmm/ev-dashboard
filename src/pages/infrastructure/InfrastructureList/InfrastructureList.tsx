import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import bem from 'easy-bem';
import { toJS } from 'mobx';
import { Button, Form, Row, Tooltip } from 'antd';
import { Link } from 'react-router-dom';
import { observer } from 'mobx-react-lite';

import { useTableFilter } from '~/shared/hooks';
import { IColumn } from '~/pages/merchants/interfaces';
import { add, deleteIcon, editColor, inactive, infoCircle, search, status } from '~/assets/images';
import { IInfrastructure } from '~/pages/infrastructure/interfaces';
import { infrastructureStore } from '~/shared/api/store';
import { FormField, TableComponent } from '~/shared/ui';
import { apiImageURL } from '~/shared/utils/config';
import './InfrastructureList.scss';

const InfrastructureList = observer(() => {
  const b = bem('InfrastructureList');
  const { t } = useTranslation();
  const { infrastructure, infrastructurePagination, infrastructureLoading } =
    toJS(infrastructureStore);
  const {
    filters,
    handleSearchChange,
    pagePrevHandler,
    pageNextHandler,
    changeShowByHandler,
    onChangePageCheckHandler,
  } = useTableFilter(infrastructureStore.fetchInfrastructure.bind(infrastructureStore));
  const [selectedRowKeys, setSelectedRowKeys] = useState<React.Key[]>([]);

  const onSelectChange = (selectedRowKeys: React.Key[]) => {
    setSelectedRowKeys(selectedRowKeys);
  };

  const columns: IColumn[] = [
    {
      title: 'â„–',
      width: 10,
      dataIndex: 'id',
    },
    {
      title: t('infrastructure.naming') + ' KG',
      render: (record: IInfrastructure) => {
        return (
          <div className={b('infrastructure')}>
            <img
              src={apiImageURL + record?.icon_path.replace('/app', '')}
              className={b('icon-infrastructure')}
              alt='icon'
            />
            {record?.title_ky}
          </div>
        );
      },
    },
    {
      title: t('infrastructure.naming') + ' RU',
      render: (record: IInfrastructure) => {
        return (
          <div className={b('infrastructure')}>
            <img
              src={apiImageURL + record?.icon_path.replace('/app', '')}
              className={b('icon-infrastructure')}
              alt='icon'
            />
            {record?.title_ru}
          </div>
        );
      },
    },
    {
      title: t('infrastructure.naming') + ' ENG',
      render: (record: IInfrastructure) => {
        return (
          <div className={b('infrastructure')}>
            <img
              src={apiImageURL + record?.icon_path.replace('/app', '')}
              className={b('icon-infrastructure')}
              alt='icon'
            />
            {record?.title_en}
          </div>
        );
      },
    },
    {
      title: t('merchants.status'),
      dataIndex: 'active',
      render: (text: boolean) => {
        return <img className={b('center-block')} src={text ? status : inactive} alt='status' />;
      },
    },
    {
      title: t('merchants.action'),
      render: () => {
        return (
          <div className={b('tags-block')}>
            <Tooltip
              color='#707A94'
              placement='left'
              title={
                <div className={b('info')}>
                  <img src={infoCircle} alt='infoCircle' />
                  <p>{t('tags.edit_tag')}</p>
                </div>
              }
            >
              <Button className={b('add-button')} icon={<img src={editColor} alt='plus' />} />
            </Tooltip>
            <Tooltip
              color='#707A94'
              placement='left'
              title={
                <div className={b('info')}>
                  <img src={infoCircle} alt='infoCircle' />
                  <p>{t('tags.delete_tag')}</p>
                </div>
              }
            >
              <Button className={b('delete-button')} icon={<img src={deleteIcon} alt='plus' />} />
            </Tooltip>
          </div>
        );
      },
    },
  ] as IColumn[];

  const rowSelection = {
    selectedRowKeys,
    onChange: onSelectChange,
  };

  return (
    <Row justify='space-between' data-testid='infrastructure-component' className={b()}>
      <Row className={b('search-pagination-block')}>
        <div>
          <Form>
            <FormField
              style={{ width: 300 }}
              id='autocomplete_id'
              placeholder={t('infrastructure.search')}
              size='large'
              prefix={<img src={search} alt='search' />}
              onChange={handleSearchChange}
            />
          </Form>
        </div>
      </Row>

      <Row className={b('table-block')}>
        <Link to='/infrastructure/create-infrastructure' className={b('add-block')}>
          <Button className={b('button-style')} type='primary' icon={<img src={add} alt='add' />}>
            {t('infrastructure.add_infrastructure')}
          </Button>
        </Link>

        <TableComponent
          rowKey={(record) => record.id.toString()}
          rowSelection={rowSelection}
          loading={infrastructureLoading}
          data={infrastructure || []}
          columns={columns}
          pagePrevHandler={pagePrevHandler}
          pageNextHandler={pageNextHandler}
          changeShowByHandler={changeShowByHandler}
          onChangePageCheckHandler={onChangePageCheckHandler}
          defaultSizeValue={filters?.page}
          pages={infrastructurePagination}
        />
      </Row>
    </Row>
  );
});

export default InfrastructureList;
