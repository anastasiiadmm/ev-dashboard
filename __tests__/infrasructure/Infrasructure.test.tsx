import React from 'react';
import { act, render } from '@testing-library/react';
import "@testing-library/jest-dom";
import { BrowserRouter } from 'react-router-dom';

import '../../__mocks__/react-i18next.mock';
import '../../__mocks__/matchMedia.mock';
import '../../__mocks__/i18nextMock';
import InfrastructureList from "../../src/pages/infrastructure/InfrastructureList/InfrastructureList";

const mockInfrastructure = [
  {
    id: 6,
    icon_path: '/app/media/surrounding/pharmacy.png',
    title_ru: 'Аптека',
    title_en: 'Аптека',
    title_ky: 'Аптека',
    active: true,
  }
];

beforeAll(() => {
  process.env.NODE_ENV = 'development';
  jest.mock("~/shared/api/store", () => ({
    infrastructureStore: {
      fetchInfrastructure: jest.fn(() => Promise.resolve(mockInfrastructure)),
      infrastructure: mockInfrastructure,
    },
  }));
});

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: key => key })
}));

jest.mock('~/shared/utils/config', () => ({
  apiURL: 'http://localhost:8000',
}));

afterEach(() => {
  jest.restoreAllMocks();
});

describe('Infrastructure UI Component', () => {
  test('renders Infrastructure data table correctly', async () => {
    const { getByTestId } = render(
      <BrowserRouter>
        <InfrastructureList />
      </BrowserRouter>
    );

    await act(async () => {
      expect(getByTestId('infrastructure-component')).toBeInTheDocument();
    });
  });
});
