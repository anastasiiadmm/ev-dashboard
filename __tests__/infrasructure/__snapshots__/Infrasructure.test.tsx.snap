// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Infrastructure UI Component Render component toMatchSnapshot() 1`] = `
<DocumentFragment>
  <div
    class="ant-row ant-row-space-between InfrastructureList css-dev-only-do-not-override-xu9wm8"
    data-testid="infrastructure-component"
  >
    <div
      class="ant-row InfrastructureList__search-pagination-block css-dev-only-do-not-override-xu9wm8"
    >
      <div>
        <form
          class="ant-form ant-form-horizontal css-dev-only-do-not-override-xu9wm8"
        >
          <div
            class="ant-form-item undefined custom-label css-dev-only-do-not-override-xu9wm8"
            style="width: 300px;"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-xu9wm8"
            >
              <div
                class="ant-col ant-form-item-control css-dev-only-do-not-override-xu9wm8"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <span
                      class="ant-input-affix-wrapper css-dev-only-do-not-override-xu9wm8"
                    >
                      <span
                        class="ant-input-prefix"
                      >
                        <img
                          alt="search"
                          src="[object Object]"
                        />
                      </span>
                      <input
                        class="ant-input css-dev-only-do-not-override-xu9wm8"
                        placeholder="infrastructure.search"
                        type="text"
                        value=""
                      />
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
    <div
      class="ant-row InfrastructureList__table-block css-dev-only-do-not-override-xu9wm8"
    >
      <button
        class="ant-btn css-dev-only-do-not-override-xu9wm8 ant-btn-primary InfrastructureList__button-style"
        type="button"
      >
        <span
          class="ant-btn-icon"
        >
          <img
            alt="add"
            src="[object Object]"
          />
        </span>
        <span>
          infrastructure.add_infrastructure
        </span>
      </button>
      <div
        class="TableComponent"
      >
        <div
          class="ant-table-wrapper css-dev-only-do-not-override-xu9wm8"
        >
          <div
            class="ant-spin-nested-loading css-dev-only-do-not-override-xu9wm8"
          >
            <div
              class="ant-spin-container"
            >
              <div
                class="ant-table ant-table-empty ant-table-scroll-horizontal"
              >
                <div
                  class="ant-table-container"
                >
                  <div
                    class="ant-table-content"
                    style="overflow-x: auto; overflow-y: hidden;"
                  >
                    <table
                      style="width: 950px; min-width: 100%; table-layout: auto;"
                    >
                      <colgroup>
                        <col
                          class="ant-table-selection-col"
                        />
                        <col
                          style="width: 10px;"
                        />
                      </colgroup>
                      <thead
                        class="ant-table-thead"
                      >
                        <tr>
                          <th
                            class="ant-table-cell ant-table-selection-column"
                            scope="col"
                          >
                            <div
                              class="ant-table-selection"
                            >
                              <label
                                class="ant-checkbox-wrapper ant-checkbox-wrapper-disabled css-dev-only-do-not-override-xu9wm8"
                              >
                                <span
                                  class="ant-checkbox ant-wave-target css-dev-only-do-not-override-xu9wm8 ant-checkbox-disabled"
                                >
                                  <input
                                    aria-label="Select all"
                                    class="ant-checkbox-input"
                                    disabled=""
                                    type="checkbox"
                                  />
                                  <span
                                    class="ant-checkbox-inner"
                                  />
                                </span>
                              </label>
                            </div>
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            №
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            infrastructure.naming KG
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            infrastructure.naming RU
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            infrastructure.naming ENG
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            merchants.status
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            merchants.action
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        class="ant-table-tbody"
                      >
                        <tr
                          aria-hidden="true"
                          class="ant-table-measure-row"
                          style="height: 0px; font-size: 0px;"
                        >
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                        </tr>
                        <tr
                          class="ant-table-placeholder"
                        >
                          <td
                            class="ant-table-cell"
                            colspan="7"
                          >
                            <div
                              class="css-dev-only-do-not-override-xu9wm8 ant-empty"
                            >
                              <div
                                class="ant-empty-image"
                                style="height: 170px;"
                              >
                                <img
                                  alt="notFoundImages"
                                  src="[object Object]"
                                  style="width: 161px; height: 161px;"
                                />
                              </div>
                              <div
                                class="ant-empty-description"
                              >
                                <h2>
                                  errors.no_data_available
                                </h2>
                              </div>
                            </div>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="PaginationComponent"
        >
          <div
            style="display: inline-flex; align-items: center;"
          >
            <p
              class="PaginationComponent__title"
            >
              table.show
            </p>
             
            <div
              class="ant-form-item undefined custom-label css-dev-only-do-not-override-xu9wm8"
            >
              <div
                class="ant-row ant-form-item-row css-dev-only-do-not-override-xu9wm8"
              >
                <div
                  class="ant-col ant-form-item-control css-dev-only-do-not-override-xu9wm8"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <div
                        class="ant-select ant-select-in-form-item css-dev-only-do-not-override-xu9wm8 ant-select-single ant-select-show-arrow"
                        style="width: 65px;"
                      >
                        <div
                          class="ant-select-selector"
                        >
                          <span
                            class="ant-select-selection-search"
                          >
                            <input
                              aria-autocomplete="list"
                              aria-controls="rc_select_TEST_OR_SSR_list"
                              aria-expanded="false"
                              aria-haspopup="listbox"
                              aria-owns="rc_select_TEST_OR_SSR_list"
                              autocomplete="off"
                              class="ant-select-selection-search-input"
                              id="rc_select_TEST_OR_SSR"
                              readonly=""
                              role="combobox"
                              style="opacity: 0;"
                              type="search"
                              unselectable="on"
                              value=""
                            />
                          </span>
                          <span
                            class="ant-select-selection-item"
                            title="10"
                          >
                            10
                          </span>
                        </div>
                        <span
                          aria-hidden="true"
                          class="ant-select-arrow"
                          style="user-select: none;"
                          unselectable="on"
                        >
                          <span
                            aria-label="down"
                            class="anticon anticon-down ant-select-suffix"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="down"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                              />
                            </svg>
                          </span>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="PaginationComponent__num-block"
          >
            <p
              class="PaginationComponent__title"
            >
              table.page
            </p>
            <form
              class="ant-form ant-form-horizontal css-dev-only-do-not-override-xu9wm8"
            >
              <div
                class="ant-form-item undefined custom-label css-dev-only-do-not-override-xu9wm8"
              >
                <div
                  class="ant-row ant-form-item-row css-dev-only-do-not-override-xu9wm8"
                >
                  <div
                    class="ant-col ant-form-item-control css-dev-only-do-not-override-xu9wm8"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          class="ant-input css-dev-only-do-not-override-xu9wm8 PaginationComponent__input-style"
                          id="size"
                          name="size"
                          type="text"
                          value="1"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </form>
            <p
              class="PaginationComponent__title"
            >
              table.from 0
            </p>
          </div>
          <div>
            <button
              class="ant-btn css-dev-only-do-not-override-xu9wm8 ant-btn-primary ant-btn-icon-only"
              disabled=""
              style="margin-right: 8px;"
              type="button"
            >
              <span
                class="ant-btn-icon"
              >
                <img
                  alt="chevronLeft"
                  src="[object Object]"
                />
              </span>
            </button>
            <button
              class="ant-btn css-dev-only-do-not-override-xu9wm8 ant-btn-primary ant-btn-icon-only"
              disabled=""
              type="button"
            >
              <span
                class="ant-btn-icon"
              >
                <img
                  alt="chevronRight"
                  src="[object Object]"
                />
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
